src/libraries/System.Text.Json/ref/System.Text.Json.cs

```diff
@@ -1055,7 +1055,7 @@ public sealed partial class JsonCollectionInfoValues<TCollection>
         public JsonCollectionInfoValues() { }
         public System.Text.Json.Serialization.Metadata.JsonTypeInfo ElementInfo { get { throw null; } init { } }
         public System.Text.Json.Serialization.Metadata.JsonTypeInfo? KeyInfo { get { throw null; } init { } }
-        public System.Text.Json.Serialization.JsonNumberHandling NumberHandling { get { throw null; } init { } }
+        public System.Text.Json.Serialization.JsonNumberHandling? NumberHandling { get { throw null; } init { } }
```

Comment:
This not being nullable initially was an oversight - we need this to distinguish between "Strict" being specified and nothing being specified so that the right precedence can be honored (setting on prop wins over type; setting on type wins over options etc.). This is in line with `Metadata.JsonPropertyInfoValues.NumberHandling`.

Reply:
`JsonNumberHandling` is an enum so making it nullable would be a binary breaking change. We can't do that.

Reply:
I don't believe there's another way to fix the numerous bugs (and test failures) that are caused by this oversight. Can we take the break now since it's early in the release and give advance notice?

Reply:
There's not a lot we can do here, unfortunately. Changing the signature like that would break binary compatibility, resulting in code compiled against `net7.0` to throw `MissingMemberException` when run against `net8.0`. 

See also https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/breaking-changes.md. This change clearly falls under Bucket 1.

Why can't the source generator simply use the [`JsonTypeInfo.NumberHandling` property](https://docs.microsoft.com/en-us/dotnet/api/system.text.json.serialization.metadata.jsontypeinfo.numberhandling?view=net-7.0#system-text-json-serialization-metadata-jsontypeinfo-numberhandling) directly?

Reply:
I agree with Eirik on using JsonTypeInfo.NumberHandling directly, we can't change this because that would be binary breaking change. I.e. if you write 6.0 lib, generate some code with JSON and now run your app on net7.0 and target 6.0 lib it will fail to load after this change

Reply:
Thanks folks. Reverted the breaking change; using the new contract API works.